config/database_test.go

```diff
@@ -467,12 +466,31 @@ func TestDatabaseStoreSet(t *testing.T) {
 
 		_, err = ds.Set(newCfg)
 		if assert.Error(t, err) {
```

Comment:
I think `require.Error` makes more sense here. That way you don't need the `if` at all, and there is no reason to continue testing if the `Set` call is not failing, because is the main check of this test.